src/OrbitBase/include/OrbitBase/File.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_FILE_H_
+#define ORBIT_BASE_FILE_H_
+
+#include <absl/strings/str_format.h>
+#include <fcntl.h>
+
+#include <filesystem>
+#include <string_view>
+
+#include "OrbitBase/Result.h"
+#include "OrbitBase/SafeStrerror.h"
+#include "OrbitBase/UniqueResource.h"
+
+#if defined(__linux)
+#include <unistd.h>
+#elif defined(_WIN32)
+#include <io.h>
+// Windows does not have TEMP_FAILURE_RETRY - define a shortcut
+#define TEMP_FAILURE_RETRY(expression) (expression)
+#endif
+
+namespace orbit_base {
+
+using unique_fd = unique_resource<int, std::function<void(int)>>;
```

Comment:
Why is snake_case being used for these types instead of PascalCase? It seems to be in clear constrast with the style guide.

Reply:
same reason why unique_resource is snake-case (as well as absl::bit_cast<>)

Reply:
And what's the reason why `unique_resource` is snake_case?

Reply:
https://google.github.io/styleguide/cppguide.html#Exceptions_to_Naming_Rules (stl-like exception)

Reply:
Alright.